python/lsst/daf/butler/registry/interfaces/_database.py

```diff
@@ -566,27 +566,33 @@ def _convertTableSpec(self, name: str, spec: ddl.TableSpec, metadata: sqlalchemy
         """
         name = self._mangleTableName(name)
         args = [self._convertFieldSpec(name, fieldSpec, metadata) for fieldSpec in spec.fields]
+        # Track indexes added for primary key and unique constraints, to make
+        # sure we don't add duplicate explicit or foreign key indexes for
+        # those.
+        allIndexes = {tuple(fieldSpec.name for fieldSpec in spec.fields if fieldSpec.primaryKey)}
         args.extend(
             sqlalchemy.schema.UniqueConstraint(
                 *columns,
                 name=self.shrinkDatabaseEntityName("_".join([name, "unq"] + list(columns)))
             )
-            for columns in spec.unique if columns not in spec.indexes
+            for columns in spec.unique
```

Comment:
I don't think so - at present, it's basically saying that if someone defines a unique constraint that's identical to the primary key, that's a bug they should fix because both of those are explicit choices (as opposed to automatically indexing FKs, which is not).